src/Components/Aspire.OpenAI/AspireOpenAIExtensions.cs

```diff
@@ -0,0 +1,124 @@
+    /// <remarks>Reads the configuration from "Aspire.OpenAI" section.</remarks>
+    public static void AddOpenAIClient(
+        this IHostApplicationBuilder builder,
+        string connectionName,
+        Action<OpenAISettings>? configureSettings = null,
+        Action<OpenAIClientOptions>? configureOptions = null)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+        ArgumentNullException.ThrowIfNull(connectionName);
+
+        AddOpenAIClient(builder, DefaultConfigSectionName, configureSettings, configureOptions, connectionName, serviceKey: null);
+    }
+
+    /// <summary>
+    /// Registers <see cref="OpenAIClient"/> as a singleton for given <paramref name="name"/> in the services provided by the <paramref name="builder"/>.
+    /// </summary>
+    /// <param name="builder">The <see cref="IHostApplicationBuilder" /> to read config from and add services to.</param>
+    /// <param name="name">The name of the component, which is used as the <see cref="ServiceDescriptor.ServiceKey"/> of the service and also to retrieve the connection string from the ConnectionStrings configuration section.</param>
+    /// <param name="configureSettings">An optional method that can be used for customizing the <see cref="OpenAISettings"/>. It's invoked after the settings are read from the configuration.</param>
+    /// <param name="configureOptions">An optional method that can be used for customizing the <see cref="OpenAIClientOptions"/>.</param>
+    /// <remarks>Reads the configuration from "Aspire.OpenAI:{name}" section.</remarks>
+    public static void AddKeyedOpenAIClient(
+        this IHostApplicationBuilder builder,
+        string name,
+        Action<OpenAISettings>? configureSettings = null,
+        Action<OpenAIClientOptions>? configureOptions = null)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+        ArgumentException.ThrowIfNullOrEmpty(name);
+
+        AddOpenAIClient(builder, $"{DefaultConfigSectionName}:{name}", configureSettings, configureOptions, connectionName: name, serviceKey: name);
+    }
+
+    private static void AddOpenAIClient(
+        this IHostApplicationBuilder builder,
+        string configurationSectionName,
+        Action<OpenAISettings>? configureSettings,
+        Action<OpenAIClientOptions>? configureOptions,
+        string connectionName,
+        string? serviceKey)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+
+        var settings = new OpenAISettings();
+        builder.Configuration.GetSection(configurationSectionName).Bind(settings);
+
+        if (builder.Configuration.GetConnectionString(connectionName) is string connectionString)
+        {
+            settings.ParseConnectionString(connectionString);
+        }
+
+        configureSettings?.Invoke(settings);
+
+        var options = new OpenAIClientOptions();
+        builder.Configuration.GetSection($"{configurationSectionName}:ClientOptions").Bind(options);
```

Comment:
I think we should use the "IOptions" pattern here, similar to how we do it in Redis:

https://github.com/dotnet/aspire/blob/0490bcbc4318c22774d162b3f49382f148cfa6b2/src/Components/Aspire.StackExchange.Redis/AspireRedisExtensions.cs#L89-L96

We are probably inconsistent in the other client libraries, but Redis and the Azure ones follow IOptions pattern.